src/Tribe/Aggregator/Event.php

```diff
@@ -231,11 +231,10 @@ public static function get_post_by_meta( $key = 'global_id', $value = null ) {
 			'global_id_lineage' => self::$global_id_lineage_key,
 		);
 
-		if ( ! isset( $keys[ $key ] ) ) {
-			return false;
+		if ( isset( $keys[ $key ] ) ) {
+			$key = $keys[ $key ];
```

Comment:
I've pushed code to simply require the `$key` parameter.
We always use it with a key.